### ISSUE TITLE
Fix inability to submit password in Firefox

### ISSUE THREAD

**Issue by timstackblock** [blockstack/blockstack-app#1750]
@markmhx  last week or so you asked about an infinite spinner when attempting to log into the web wallet which was reported by a user in one of our channels.
I see this on the initial download of the web wallet in FF, I am guessing that I had a previous version of the wallet installed and when I downloaded the latest version it was installed but opened locked showing the lock screen below. Then it prompted me for a password to unlock the wallet and an infinite spinner appeared. 

<img width="1404" alt="Screen Shot 2021-09-27 at 4 15 28 PM" src="https://user-images.githubusercontent.com/49165468/134979918-cb629394-db99-4d71-8562-d5b6baf26205.png">

I cannot reproduce this in chrome
new downloads go right to this page

<img width="1342" alt="Screen Shot 2021-09-27 at 4 18 05 PM" src="https://user-images.githubusercontent.com/49165468/134979963-9dafb94f-d246-4e5f-ba13-b0b2ed0da8ff.png">

The work around in FF is clicking the sign out button if you encounter this issue
Then the locked screen works as expected.

**Comment by kyranjamie** [comment]
Are you able to reliably replicate this issue, @timstackblock. I'm struggling to.

**Edit:** I just updated the extension `2.19.0` --> `2.19.1` and was able to replicate this. 

To replicate locally:
1) Build a version of the wallet
2) Duplicate the `dist` folder and call `dist-old`
3) Update version in `package.json`
4) Build another version
5) Load in `dist-old` version of the first extension instance & create wallet
6) Load in `dist` and try signing in

![image](https://user-images.githubusercontent.com/1618764/135421986-b23d9aba-a128-4732-8411-afd9a71f2d39.png)

Background scripts errors with:

```
Unchecked lastError value: Error: Promised response from onMessage listener went out of scope
```

**Comment by timstackblock** [comment]
I think the sequence that you explained happens naturally in FF when someone downloads the new version of the wallet if there are remnants of the old version still connected to the browser.

In Chrome I can run 2 versions of the extension side by side without an issue, chrome treats the new version and the old version as separate extensions. I usually delete the old version but I noticed this while installing the new version of the extension on a different machine I hadn't use in a a while.

**Comment by kyranjamie** [comment]
> Chrome I can run 2 versions of the extension side by side without an issue

Is that so? 🤔  We have a separate dev version, but you should only be able to run 1 production, and 1 dev instance at the same time.

I have a feeling this issue is something the same as this bug report https://bugzilla.mozilla.org/show_bug.cgi?id=1643186. There's some behaviour that happens on extension update, that likely needs to be handled.

**Comment by timstackblock** [comment]
Yes thats what I was referring to the dev version I loaded manually and the prod version I downloaded they can both run at the same time without issue in chrome. But the test dev version is what I had loaded in FF and when the prod version was download and installed this issue happen organically.

**Comment by markmhendrickson** [comment]
Either similar or related: https://github.com/hirosystems/stacks-wallet-web/issues/2014

**Comment by kyranjamie** [comment]
@markmhx this issue is unrelated to slow performance. This is a case of it being completely broken.

**Comment by pors** [comment]
I also have this issue, some details:

- FF 95.0 (64-bit) on M1 Mac
- Extension 2.23.2 (but had the issue with older versions as well)

When I had the issue it kept on spinning after entering the password. I tried at least 10 times. Opening and closing FF didn't make a difference.
A message in the console: `Content Security Policy: Ignoring “'unsafe-inline'” within script-src or style-src: nonce-source or hash-source specified`.

I resolved it by logging out of the wallet, entering the HD wallet phrase again. This time no more problems. After closing/opening FF still no problems.
10 minutes later after another FF close/open the problem is back... 
Recreating the wallet solves it again.

**Comment by fbwoolf** [comment]
I have tried to repro this but can't so far. Will keep trying.

**Comment by pors** [comment]
Per request by @kyranjamie a screen video. I can reproduce it very easy and the spinner goes on forever. The only way to resolve it for me it to close/open FF a couple of times and get lucky.

No errors or warnings in the console.

https://user-images.githubusercontent.com/1078320/145705914-743fe73e-d044-473e-93b8-0c22fe8b1d73.mov

Let me know if I can do anything else.

**Comment by markmhendrickson** [comment]
@pors Is this still an active issue for you?

**Comment by pors** [comment]
Yes it is. I stopped using it with FF for this reason, but also with version v3.7.2 it loads forever after entering my password. The only way to get it to log in is to "Sign out" and recreate the wallet with the phrase.

Maybe it has to do with my weird password `*a._x4FUEVaiva*`? (I changed it now :))
